names.c

```diff
@@ -21,7 +21,9 @@
 #include <stdio.h>
 #include <ctype.h>
 
-#include <libudev.h>
+#ifdef HAVE_UDEV
```

Comment:
putting #ifdef in .c files like this quickly turns this into a maintance nightmare.

if you really want to do this, put it in a .h file like Linux does, and provide versions with, and without HAVE_UDEV present.  That way the .c code does not care and it is easier to follow and maintain over time.